nlp/language_modeling/conf/megatron_gpt_prune.yaml

```diff
@@ -0,0 +1,47 @@
+inference:
+  greedy: false # Whether or not to use sampling ; use greedy decoding otherwise
+  top_k: 0  # The number of highest probability vocabulary tokens to keep for top-k-filtering.
+  top_p: 0.9 # If set to float < 1, only the most probable tokens with probabilities that add up to top_p or higher are kept for generation.
+  temperature: 1.0 # sampling temperature
+  add_BOS: true # add the bos token at the begining of the prompt
+  tokens_to_generate: 30 # The minimum length of the sequence to be generated.
+  all_probs: false  # whether return the log prob for all the tokens in vocab
+  repetition_penalty: 1.2  # The parameter for repetition penalty. 1.0 means no penalty.
+  min_tokens_to_generate: 0  # The minimum length of the sequence to be generated.
+  compute_logprob: false  # a flag used to compute logprob of all the input text, a very special case of running inference, default False
+  batch_size: 64 # batch size for inference
+  max_context_length: 512 # max length of the context, input sequence will be truncated if it is longer than this
+
+trainer:
+  devices: 1
+  num_nodes: 1
+  accelerator: gpu
+  logger: false # logger provided by exp_manager
+  precision: bf16 # 16, 32, or bf16
+  enable_checkpointing: false
+
+model:
+  tensor_model_parallel_size: 1
+  pipeline_model_parallel_size: 1
+  restore_from_path: llama3-8b-base.nemo # Nemo file path
+
+  ## Activation Checkpoint
+  activations_checkpoint_granularity: null # 'selective' or 'full'
+  activations_checkpoint_method: null # 'uniform', 'block', not used with 'selective'
+
+prune:
+  decoder_type: ${export.decoder_type} # gptnext, gpt2, llama
+  calib_dataset: cnn_dailymail # wikitext, cnn_dailymail, or a local dataset
+  num_calib_size: 512 # number of samples used for calibration
+  ffn_hidden_size: 3584 # ffn_hidden_size in the pruned model, ffn_hidden_size // 4
+  num_attention_heads: 8 # num_attention_heads in the pruned model, num_attention_heads // 4
+  num_query_groups: 4 # num_query_groups in the pruned model, num_query_groups // 2
+
+export:
+  decoder_type: llama # gptnext, gpt2, llama
+  inference_tensor_parallel: 1 # Default using 1 TP for inference
+  inference_pipeline_parallel: 1 # Default using 1 PP for inference
+  dtype: ${trainer.precision} # Default precision data type
+  save_path: llama3-8b-base-pruned.nemo # Path where the pruned model will be saved
+  compress: false # Whether save_path should be a tarball or a directory
+  sample_output: true # Whether to run a sample prompt before saving
```

Comment:
```suggestion
model:
  tensor_model_parallel_size: 1 # can only be 1 for now
  pipeline_model_parallel_size: 1
  sequence_parallel: False # should be disabled during pruning
  restore_from_path: ???

prune:
  calib_dataset: cnn_dailymail # wikitext, cnn_dailymail, or a local dataset
  num_calib_size: 1024 # number of samples used for calibration
  save_path: ??? # Path where the pruned model will be saved
  ffn_hidden_size: null # ffn_hidden_size in the pruned model
  num_attention_heads: null # num_attention_heads in the pruned model
  num_query_groups: null # num_query_groups in the pruned model
```

Reply:
The values in this file are all default values. So we should provide a working value set. They can be overwritten through command line arguments.

Reply:
Thats the convension in most config files with the exception of the ptq script.
- https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/conf/megatron_llama_distill.yaml
- https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/conf/megatron_llama_config.yaml
- https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/conf/megatron_gpt_config.yaml

I think these paths for loading and saving are always going to changed based on the user and the pruning parameters are also model dependent. Having them default to something might cause unexpected behavior if users don't overwrite them for their model and they wont get any errors as well

Reply:
I believe it is fine and much better to have the values set because we provide this setting:

`  restore_from_path: llama3-8b-base.nemo # Nemo file path`

meaning the default config is for this model and we have tested the setting for this particular model. Users can just grab and place the model here and everything should just work.

a ? or a ?? provide no info for the user.